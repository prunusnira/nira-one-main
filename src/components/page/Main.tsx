import Item from "../item/item";
import {
    Desc,
    Title,
    BWSpan,
    TitleSub,
    Wrapper,
    WBSpan,
    DescCenter,
} from "../common.style";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faAnglesDown } from "@fortawesome/free-solid-svg-icons";

type Props = {
    bgImage: string;
};

const Main = ({ bgImage }: Props) => {
    return (
        <Item bgImage={bgImage}>
            <Wrapper>
                <Title>
                    <BWSpan>nira.one</BWSpan>
                </Title>
                <TitleSub>
                    <WBSpan>GATEWAY OF NIRA.ONE SERVICES</WBSpan>
                </TitleSub>
                <Desc>
                    Skill Navigator
                    <br />
                    PIU Manager
                    <br />
                    Twitch BanPicker
                    <br />
                    Portfolio
                    <br />
                    Others
                </Desc>
                <DescCenter>
                    <span>Scroll Down</span>
                    <FontAwesomeIcon icon={faAnglesDown} />
                </DescCenter>
            </Wrapper>
        </Item>
    );
};

export default Main;
