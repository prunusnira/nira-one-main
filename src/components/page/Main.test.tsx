import { render } from "@testing-library/react";
import Main from "./Main";

describe("main section", () => {
    it("snapshot test", () => {
        const wrapper = render(<Main bgImage="" />);
        expect(wrapper).toMatchSnapshot();
    });

    it("render test", () => {
        const wrapper = render(<Main bgImage="" />);
        wrapper.getByText("nira.one");
    });
});
