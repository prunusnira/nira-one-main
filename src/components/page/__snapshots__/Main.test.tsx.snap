// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`main section render test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="target css-1hzir52"
      >
        <div
          class="css-15z58y6"
        >
          <div
            class="css-h2mica"
          >
            <span
              class="css-1d7eyad"
            >
              nira.one
            </span>
          </div>
          <div
            class="css-wyy442"
          >
            <span
              class="css-1fqqz26"
            >
              GATEWAY OF NIRA.ONE SERVICES
            </span>
          </div>
          <div
            class="css-mwv1s8"
          >
            Skill Navigator
            <br />
            PIU Manager
            <br />
            Twitch BanPicker
            <br />
            Portfolio
            <br />
            Others
          </div>
          <div
            class="css-17ohs4h"
          >
            <span>
              Scroll Down
            </span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angles-down "
              data-icon="angles-down"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M246.6 470.6c-12.5 12.5-32.8 12.5-45.3 0l-160-160c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L224 402.7 361.4 265.4c12.5-12.5 32.8-12.5 45.3 0s12.5 32.8 0 45.3l-160 160zm160-352l-160 160c-12.5 12.5-32.8 12.5-45.3 0l-160-160c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L224 210.7 361.4 73.4c12.5-12.5 32.8-12.5 45.3 0s12.5 32.8 0 45.3z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="target css-1hzir52"
    >
      <div
        class="css-15z58y6"
      >
        <div
          class="css-h2mica"
        >
          <span
            class="css-1d7eyad"
          >
            nira.one
          </span>
        </div>
        <div
          class="css-wyy442"
        >
          <span
            class="css-1fqqz26"
          >
            GATEWAY OF NIRA.ONE SERVICES
          </span>
        </div>
        <div
          class="css-mwv1s8"
        >
          Skill Navigator
          <br />
          PIU Manager
          <br />
          Twitch BanPicker
          <br />
          Portfolio
          <br />
          Others
        </div>
        <div
          class="css-17ohs4h"
        >
          <span>
            Scroll Down
          </span>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-angles-down "
            data-icon="angles-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M246.6 470.6c-12.5 12.5-32.8 12.5-45.3 0l-160-160c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L224 402.7 361.4 265.4c12.5-12.5 32.8-12.5 45.3 0s12.5 32.8 0 45.3l-160 160zm160-352l-160 160c-12.5 12.5-32.8 12.5-45.3 0l-160-160c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L224 210.7 361.4 73.4c12.5-12.5 32.8-12.5 45.3 0s12.5 32.8 0 45.3z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`main section snapshot test 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="target css-1hzir52"
      >
        <div
          class="css-15z58y6"
        >
          <div
            class="css-h2mica"
          >
            <span
              class="css-1d7eyad"
            >
              nira.one
            </span>
          </div>
          <div
            class="css-wyy442"
          >
            <span
              class="css-1fqqz26"
            >
              GATEWAY OF NIRA.ONE SERVICES
            </span>
          </div>
          <div
            class="css-mwv1s8"
          >
            Skill Navigator
            <br />
            PIU Manager
            <br />
            Twitch BanPicker
            <br />
            Portfolio
            <br />
            Others
          </div>
          <div
            class="css-17ohs4h"
          >
            <span>
              Scroll Down
            </span>
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-angles-down "
              data-icon="angles-down"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M246.6 470.6c-12.5 12.5-32.8 12.5-45.3 0l-160-160c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L224 402.7 361.4 265.4c12.5-12.5 32.8-12.5 45.3 0s12.5 32.8 0 45.3l-160 160zm160-352l-160 160c-12.5 12.5-32.8 12.5-45.3 0l-160-160c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L224 210.7 361.4 73.4c12.5-12.5 32.8-12.5 45.3 0s12.5 32.8 0 45.3z"
                fill="currentColor"
              />
            </svg>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="target css-1hzir52"
    >
      <div
        class="css-15z58y6"
      >
        <div
          class="css-h2mica"
        >
          <span
            class="css-1d7eyad"
          >
            nira.one
          </span>
        </div>
        <div
          class="css-wyy442"
        >
          <span
            class="css-1fqqz26"
          >
            GATEWAY OF NIRA.ONE SERVICES
          </span>
        </div>
        <div
          class="css-mwv1s8"
        >
          Skill Navigator
          <br />
          PIU Manager
          <br />
          Twitch BanPicker
          <br />
          Portfolio
          <br />
          Others
        </div>
        <div
          class="css-17ohs4h"
        >
          <span>
            Scroll Down
          </span>
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-angles-down "
            data-icon="angles-down"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M246.6 470.6c-12.5 12.5-32.8 12.5-45.3 0l-160-160c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L224 402.7 361.4 265.4c12.5-12.5 32.8-12.5 45.3 0s12.5 32.8 0 45.3l-160 160zm160-352l-160 160c-12.5 12.5-32.8 12.5-45.3 0l-160-160c-12.5-12.5-12.5-32.8 0-45.3s32.8-12.5 45.3 0L224 210.7 361.4 73.4c12.5-12.5 32.8-12.5 45.3 0s12.5 32.8 0 45.3z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
